src/http-blockstore.ts

```diff
@@ -0,0 +1,109 @@
+import type { Blockstore, Pair } from 'interface-blockstore'
+import type { Connection } from '@libp2p/interface-connection'
+import type {
+  AbortOptions,
+  Await,
+  AwaitIterable
+} from 'interface-store'
+import type { Libp2p } from 'libp2p'
+import type { CID } from 'multiformats/cid'
+import { fetchViaDuplex } from '@marcopolo_/libp2p-fetch'
+
+// TODOS:
+// - Validate the blocks we fetch
```

Comment:
We need to do this and it should be fairly straightforward to do (just grab the multihash, lookup an implementation of the hash function, run it over the bytes and see if they match)